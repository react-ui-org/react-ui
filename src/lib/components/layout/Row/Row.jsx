import PropTypes from 'prop-types';
import React from 'react';
import styles from './Row.scss';

const Row = (props) => {
  const {
    children,
  } = props;

  return (
    <div className={styles.row}>
      {children}
    </div>
  );
};

Row.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Row;
