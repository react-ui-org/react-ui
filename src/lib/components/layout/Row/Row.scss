@import '../../../styles/tools/offset';

.row {
  display: flex;
  flex-wrap: wrap;
  align-items: baseline;
  margin-bottom: offset(3);
}

.start,
.end {
  flex: none;
}

.end {
  padding-left: offset(3);
  margin-left: auto;
}
