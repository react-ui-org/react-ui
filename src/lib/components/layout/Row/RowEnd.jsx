import PropTypes from 'prop-types';
import React from 'react';
import styles from './Row.scss';

const RowEnd = ({ children }) => (
  <div className={styles.end}>
    {children}
  </div>
);

RowEnd.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default RowEnd;
