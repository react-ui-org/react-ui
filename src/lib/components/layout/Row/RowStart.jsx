import PropTypes from 'prop-types';
import React from 'react';
import styles from './Row.scss';

const RowStart = ({ children }) => (
  <div className={styles.start}>
    {children}
  </div>
);

RowStart.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default RowStart;
