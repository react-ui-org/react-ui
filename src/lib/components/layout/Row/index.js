export { default as Row } from './Row';
export { default as RowEnd } from './RowEnd';
export { default as RowStart } from './RowStart';
