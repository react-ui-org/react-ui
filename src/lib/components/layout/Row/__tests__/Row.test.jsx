import React from 'react';
import { shallow } from 'enzyme';
import Row from '../Row';
import RowEnd from '../RowEnd';
import RowStart from '../RowStart';

describe('rendering', () => {
  it('renders correctly with a single child', () => {
    const tree = shallow((
      <Row>
        <RowStart>start</RowStart>
      </Row>
    ));

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with multiple children', () => {
    const tree = shallow((
      <Row>
        <RowStart>start</RowStart>
        <RowEnd>end</RowEnd>
      </Row>
    ));

    expect(tree).toMatchSnapshot();
  });
});
