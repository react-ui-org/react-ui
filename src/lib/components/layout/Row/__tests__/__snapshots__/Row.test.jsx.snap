// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly with a single child 1`] = `
<div
  className="row"
>
  <RowStart>
    start
  </RowStart>
</div>
`;

exports[`rendering renders correctly with multiple children 1`] = `
<div
  className="row"
>
  <RowStart>
    start
  </RowStart>
  <RowEnd>
    end
  </RowEnd>
</div>
`;
